hbw/app/models/hbw/common/adapter.rb

```diff
@@ -110,6 +110,14 @@ def form(task_id, entity_class)
         end
       end
 
+      def form_definition(task_id, entity_class)
+        task = HBW::Task.new('id' => task_id, 'processDefinition' => nil)
+
+        HBW::Form.with_connection(api) do
+          HBW::Form.fetch(task, entity_class)
```

Comment:
Looks like you're still passing redundant data to single method used for different purposes. I don't think this kind of DRY is good in fact.